src/fprime_extras/lint/rules/rule_valid_port_types.py

```diff
@@ -0,0 +1,162 @@
+import logging
+
+from ...core.hooks import HookRegistry as hooks
+from .base import AbstractRule
+from .base import LintSeverity
+from .base import Notification
+from .registry import RuleListRegistry
+from ...core.util import find_fprime
+
+from pathlib import Path
+import os, sys
+import lxml.etree as etree
+from lxml.etree import XMLSyntaxError
+
+log = logging.getLogger(__name__)
+slug = 'invalid-port-type'
+# TODO: import following list from model instead of defining it here.
+port_kinds = ['output', 'sync_input', 'async_input', 'guarded_input']
+
+
+@RuleListRegistry.register()
+class RuleDanglingPortConnections(AbstractRule):
+    """Check if input XML file can be parsed."""
+
+    def __init__(self):
+        super().__init__()
+
+    @classmethod
+    def register_hook_callbacks(cls):
+        hooks.register('fprime_extras/lint/document_model_rules', cls())
+
+    def __call__(self, config, extras_file, doc_model):
+        notifications = []
+        fprime_path = config.fprime_root
+        if fprime_path is None or not os.path.exists(fprime_path):
+            log.info("Received input root of {} which is invalid. Searching for Fprime root instead".format(fprime_path))
+            if fprime_path is not None:
+                msg = "{} does not exist. Searching for Fprime root as parent of component.".format(fprime_path)
+                note = Notification(filename=extras_file.full_filename,
+                                    rule_slug=slug,
+                                    description=msg,
+                                    line_number=0,
+                                    column=0,
+                                    severity=LintSeverity.WARNING
+                                    )
```

Comment:
I think that because this problem is not fixed by changing the file being linted, its probably not appropriate as a ilnt warning.  Its definitely worthy of an info level log statement, which you have already provided.

Reply:
In fact, if we find the root automatically, we don't want to emit a lint notification.

Reply:
I removed this notification